modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java

```diff
@@ -292,13 +326,33 @@ private CacheConsistencyViolationEventListener(String name) {
             @Override public boolean apply(CacheConsistencyViolationEvent evt) {
                 assert evt instanceof CacheConsistencyViolationEvent;
 
-                if (!evt.getCacheName().equals(cacheName))
+                if (!cacheNames.contains(evt.getCacheName()))
                     return true; // Skipping other caches results, which are generated by concurrent executions.
 
                 evts.add(evt);
 
                 return true;
             }
         }
+
+        /**
+         *
+         */
+        private static class PerCacheBatch {
+            /** Cache. */
+            private final IgniteCache<Object, Object> cache;
+
+            /** Keys. */
+            private final Set<Object> keys;
+
+            /**
+             * @param cache Cache.
+             * @param keys Keys.
+             */
+            public PerCacheBatch(IgniteCache<Object, Object> cache, Set<Object> keys) {
```

Comment:
This constructor is always called with empty keys, seems that it should be changed a little bit : an empty hash set could be created during initialization

Reply:
Sure!